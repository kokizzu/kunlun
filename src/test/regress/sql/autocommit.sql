show autocommit;
drop schema if exists autocommit_schm cascade;
create schema autocommit_schm;
use autocommit_schm;
show search_path;
create table t1(a serial primary key,  b int);

insert into t1(b) values(17);
select*from t1;

set autocommit=off;
insert into t1(b) values(19);
commit;
-- better check before and after above commit in another session
select*from t1;

set autocommit=on;
insert into t1(b) values(31);
select*from t1;

begin;insert into t1(b) values(37);commit;
-- better check before and after above commit in another session
select*from t1;

set autocommit to off;
insert into t1(b) values(53);
commit;
-- better check before and after above commit in another session
select*from t1;

BEGIN;BEGIN;
insert into t1(b) values(59);
commit;
-- better check before and after above commit and BEGIN in another session
select*from t1;

BEGIN;BEGIN;BEGIN;
insert into t1(b) values(61);
BEGIN; -- implicitly commit
insert into t1(b) values(67);
begin; -- implicitly commit
commit;
-- better check before and after above commit and each BEGIN stmt in another session
select*from t1;

show autocommit;

insert into t1(b) values(71);

insert into t1(b) values(73);
set autocommit=true; -- implicitly commit

-- better check before and after above stmt in another session
select*from t1;

set autocommit=off;
insert into t1(b) values(79);
begin; -- implicitly commit
insert into t1(b) values(83);
set autocommit=1; -- implicitly commit
insert into t1(b) values(89);

-- better check before and after above begin and "set autocommit=1" stmt in another session
select*from t1;

drop schema if exists autocommit_schm cascade;
