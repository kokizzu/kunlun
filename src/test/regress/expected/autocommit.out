show autocommit;
 autocommit 
------------
 on
(1 row)

drop schema if exists autocommit_schm cascade;
psql:sql/autocommit.sql:2: NOTICE:  schema "autocommit_schm" does not exist, skipping
DROP SCHEMA
create schema autocommit_schm;
CREATE SCHEMA
use autocommit_schm;
USE
show search_path;
   search_path   
-----------------
 autocommit_schm
(1 row)

create table t1(a serial primary key,  b int);
CREATE TABLE
insert into t1(b) values(17);
INSERT 0 1
select*from t1;
 a | b  
---+----
 1 | 17
(1 row)

set autocommit=off;
SET
insert into t1(b) values(19);
INSERT 0 1
commit;
COMMIT
-- better check before and after above commit in another session
select*from t1;
 a | b  
---+----
 1 | 17
 2 | 19
(2 rows)

set autocommit=on;
SET
insert into t1(b) values(31);
INSERT 0 1
select*from t1;
 a | b  
---+----
 1 | 17
 2 | 19
 3 | 31
(3 rows)

begin;insert into t1(b) values(37);commit;
BEGIN
INSERT 0 1
COMMIT
-- better check before and after above commit in another session
select*from t1;
 a | b  
---+----
 1 | 17
 2 | 19
 3 | 31
 4 | 37
(4 rows)

set autocommit to off;
SET
insert into t1(b) values(53);
INSERT 0 1
commit;
COMMIT
-- better check before and after above commit in another session
select*from t1;
 a | b  
---+----
 1 | 17
 2 | 19
 3 | 31
 4 | 37
 5 | 53
(5 rows)

BEGIN;BEGIN;
BEGIN
BEGIN
insert into t1(b) values(59);
INSERT 0 1
commit;
COMMIT
-- better check before and after above commit and BEGIN in another session
select*from t1;
 a | b  
---+----
 1 | 17
 2 | 19
 3 | 31
 4 | 37
 5 | 53
 6 | 59
(6 rows)

BEGIN;BEGIN;BEGIN;
BEGIN
BEGIN
BEGIN
insert into t1(b) values(61);
INSERT 0 1
BEGIN; -- implicitly commit
BEGIN
insert into t1(b) values(67);
INSERT 0 1
begin; -- implicitly commit
BEGIN
commit;
COMMIT
-- better check before and after above commit and each BEGIN stmt in another session
select*from t1;
 a | b  
---+----
 1 | 17
 2 | 19
 3 | 31
 4 | 37
 5 | 53
 6 | 59
 7 | 61
 8 | 67
(8 rows)

show autocommit;
 autocommit 
------------
 off
(1 row)

insert into t1(b) values(71);
INSERT 0 1
insert into t1(b) values(73);
INSERT 0 1
set autocommit=true; -- implicitly commit
SET
-- better check before and after above stmt in another session
select*from t1;
 a  | b  
----+----
  1 | 17
  2 | 19
  3 | 31
  4 | 37
  5 | 53
  6 | 59
  7 | 61
  8 | 67
  9 | 71
 10 | 73
(10 rows)

set autocommit=off;
SET
insert into t1(b) values(79);
INSERT 0 1
begin; -- implicitly commit
BEGIN
insert into t1(b) values(83);
INSERT 0 1
set autocommit=1; -- implicitly commit
SET
insert into t1(b) values(89);
INSERT 0 1
-- better check before and after above begin and "set autocommit=1" stmt in another session
select*from t1;
 a  | b  
----+----
  1 | 17
  2 | 19
  3 | 31
  4 | 37
  5 | 53
  6 | 59
  7 | 61
  8 | 67
  9 | 71
 10 | 73
 11 | 79
 12 | 83
 13 | 89
(13 rows)

drop schema if exists autocommit_schm cascade;
psql:sql/autocommit.sql:66: NOTICE:  drop cascades to table t1
DROP SCHEMA
